backend/controller/scaling/local_scaling.go

```diff
@@ -0,0 +1,116 @@
+package scaling
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/alecthomas/errors"
+	"github.com/alecthomas/kong"
+
+	"github.com/TBD54566975/ftl/backend/common/log"
+	"github.com/TBD54566975/ftl/backend/common/model"
+	"github.com/TBD54566975/ftl/backend/runner"
+)
+
+var _ RunnerScaling = (*LocalScaling)(nil)
+
+type LocalScaling struct {
+	cacheDir string
+	runners  []model.RunnerKey
+
+	nextBind            *url.URL
```

Comment:
I was thinking a bit about this and there's a pattern I've used previously that is quite a bit simpler and more flexible. Let's pair on it, but basically it's to use an `*atomic.Int32` for the port assignment. This allows it to be shared between the runner and controller without locking, and you can just allocate a new port at any time.

Reply:
Thinking more, it would probably be even more preferable to create a little `PortAllocator` interface.

Reply:
Sounds good!